Sources/StreamChat/ChatClient.swift

```diff
@@ -285,6 +127,94 @@ public class ChatClient {
+        configuration.timeoutIntervalForRequest = config.timeoutIntervalForRequest
+        urlSessionConfiguration = configuration
+
+        let factory = ChatClientFactory(config: config, environment: environment)
+        var apiClientEncoder = factory.makeApiClientRequestEncoder()
+        var webSocketEncoder = factory.makeWebSocketRequestEncoder()
+        let databaseContainer = factory.makeDatabaseContainer()
+        let apiClient = factory.makeApiClient(
+            encoder: apiClientEncoder,
+            urlSessionConfiguration: configuration
+        )
+        let eventNotificationCenter = factory.makeEventNotificationCenter(
+            databaseContainer: databaseContainer,
+            currentUserId: {
+                nil
+            }
+        )
+        let messageRepository = environment.messageRepositoryBuilder(
+            databaseContainer,
+            apiClient
+        )
+        let offlineRequestsRepository = environment.offlineRequestsRepositoryBuilder(
+            messageRepository,
+            databaseContainer,
+            apiClient
+        )
+        let syncRepository = environment.syncRepositoryBuilder(
+            config,
+            activeChannelControllers,
+            activeChannelListControllers,
+            offlineRequestsRepository,
+            eventNotificationCenter,
+            databaseContainer,
+            apiClient
+        )
+        let webSocketClient = factory.makeWebSocketClient(
+            requestEncoder: webSocketEncoder,
+            urlSessionConfiguration: urlSessionConfiguration,
+            eventNotificationCenter: eventNotificationCenter
+        )
+        let connectionRepository = environment.connectionRepositoryBuilder(
+            config.isClientInActiveMode,
+            syncRepository,
+            webSocketClient,
+            apiClient,
+            environment.timerType
+        )
+        let authRepository = environment.authenticationRepositoryBuilder(
+            apiClient,
+            databaseContainer,
+            connectionRepository,
+            environment.tokenExpirationRetryStrategy,
+            environment.timerType
+        )
+
+        self.databaseContainer = databaseContainer
+        self.apiClient = apiClient
+        self.webSocketClient = webSocketClient
+        self.eventNotificationCenter = eventNotificationCenter
+        self.offlineRequestsRepository = offlineRequestsRepository
+        self.connectionRepository = connectionRepository
+        self.messageRepository = messageRepository
+        self.syncRepository = syncRepository
+        authenticationRepository = authRepository
+        extensionLifecycle = environment.extensionLifecycleBuilder(config.applicationGroupIdentifier)
+        callRepository = environment.callRepositoryBuilder(apiClient)
+        channelRepository = environment.channelRepositoryBuilder(
+            databaseContainer,
+            apiClient
+        )
+
+        authRepository.delegate = self
```

Comment:
The additional setup delegate stuff could be also moved into separate methods, since this one is becoming too big.

Reply:
I think it is useful that everything is in the same place. If I move this to a different method, I will also need to pass all the objects required for the setup 🤔 I will see if I can improve this either way

Reply:
Improved these ones a bit 👍